docs/contributing/howto.md

```diff
@@ -0,0 +1,208 @@
+
+* Use a Python virtual environment, or
+* Use Docker
+
+### a) Using a Python virtual environment for local development
+
+Install the project dependencies:
+
+```bash
+make setup
+```
+
+Activate the local python environment:
+
+```bash
+source venv/bin/activate
+```
+
+Set [Apache Airflow®](https://airflow.apache.org/) home to the ``dev/``, so you can see DAG Factory example DAGs.
+Disabled loading Airflow standard example DAGs:
+
+```bash
+export AIRFLOW_HOME=$(pwd)/dev/
+export AIRFLOW__CORE__LOAD_EXAMPLES=false
+```
+
+Then, run Airflow in standalone mode; the command below will create a new user (if it does not exist) and run the necessary Airflow component (webserver, scheduler and triggered):
+
+> Note: By default, Airflow will use sqlite as a database; you can override this by setting the variable ``AIRFLOW__DATABASE__SQL_ALCHEMY_CONN`` to the SQL connection string.
+
+```bash
+airflow standalone
+```
+
+Once Airflow is up, you can access the Airflow UI at ``http://localhost:8080``.
+
+> Note: whenever you want to start the development server, you need to activate the ``virtualenv`` and set the ``environment variables``
+
+### b) Using Docker for local development
+
+It is also possible to build the development environment using [Docker](https://www.docker.com/products/docker-desktop/):
+
+```bash
+make docker-run
+```
+
+Once the sandbox is up, you can access the Airflow UI at ``http://localhost:8080``.
+
+This approach builds a DAG Factory wheel, so if there are code changes, you'll have to stop and restart the containers:
+
+```bash
+make docker-stop
+```
+
+## 3. Testing application with hatch
+
+The tests are developed using PyTest and run using hatch.
+
+The [pyproject. toml](https://github.com/astronomer/dag-factory/blob/main/pyproject.toml) file currently defines a matrix of supported versions of Python and Airflow against which a user can run the tests.
+
+### a) Running unit tests
+
+For instance, to run unit tests using Python 3.10 and Airflow 2.5, use the following:
+
+```bash
+hatch run tests.py3.10-2.5:test-cov
+```
+
+It is also possible to run the tests using all the matrix combinations, by using:
+
+```bash
+hatch run tests:test-cov
+```
+
+### b) Running integration tests
+
+> Note: these tests will create local Python virtual environments in a hatch-managed directory
+> They will also use the user-defined `AIRFLOW_HOME`, overriding any pre-existing `airflow.cfg` and `airflow.db` files
+
+First, set the following environment variables:
+
+```bash
+export AIRFLOW_HOME=$(pwd)/dev/
+export CONFIG_ROOT_DIR=`pwd`"/dev/dags"
+export PYTHONPATH=dev/dags:$PYTHONPATH
+```
+
+To run the integration tests using Python 3.9 and Airflow 2.9, use
+
+```bash
+hatch run tests.py3.9-2.9:test-integration-setup
+hatch run tests.py3.9-2.9:test-integration
+```
+
+## 4. Pre-Commit and linting
+
+We use pre-commit to run several checks on the code before committing. To install pre-commit, run:
```

Comment:
```suggestion
We use pre-commit to run several checks on the code before committing. To install pre-commit hooks, run:
```
Additionally, we could maybe also add a prior step saying they could install pre-commit with `pip install  pre-commit`